src/main/java/com/puppycrawl/tools/checkstyle/filters/LoadPatchFileUtils.java

```diff
@@ -80,26 +85,38 @@ public String getFileName() {
             for (HunkHeader hunkHeader : fileHeader.getHunks()) {
                 final EditList edits = hunkHeader.toEditList();
                 for (Edit edit : edits) {
-                    if (NEWLINE.equals(strategy)) {
-                        if (Edit.Type.INSERT.equals(edit.getType())) {
-                            final List<Integer> lineRange = new ArrayList<>();
-                            lineRange.add(edit.getBeginB());
-                            lineRange.add(edit.getEndB());
-                            lineRangeList.add(lineRange);
-                        }
-                    }
-                    else {
-                        if (Edit.Type.INSERT.equals(edit.getType())
-                                || Edit.Type.REPLACE.equals(edit.getType())) {
-                            final List<Integer> lineRange = new ArrayList<>();
-                            lineRange.add(edit.getBeginB());
-                            lineRange.add(edit.getEndB());
-                            lineRangeList.add(lineRange);
-                        }
-                    }
+                    addSingleLineRange(lineRangeList, edit);
                 }
             }
         }
         return lineRangeList;
     }
+
+    private void addSingleLineRange(List<List<Integer>> lineRangeList, Edit edit) {
+        if (NEW_LINE.equals(strategy)) {
+            if (Edit.Type.INSERT.equals(edit.getType())) {
+                final List<Integer> lineRange = new ArrayList<>();
+                lineRange.add(edit.getBeginB());
+                lineRange.add(edit.getEndB());
+                lineRangeList.add(lineRange);
+            }
+        }
+        else if (PATCHED_LINE.equals(strategy)) {
+            if (Edit.Type.INSERT.equals(edit.getType())
+                    || Edit.Type.REPLACE.equals(edit.getType())) {
+                final List<Integer> lineRange = new ArrayList<>();
+                lineRange.add(edit.getBeginB());
+                lineRange.add(edit.getEndB());
+                lineRangeList.add(lineRange);
+            }
+        }
+        else if (Edit.Type.INSERT.equals(edit.getType())
+                || Edit.Type.REPLACE.equals(edit.getType())
+                || Edit.Type.DELETE.equals(edit.getType())) {
```

Comment:
This change is significant. We need few more tests. Can you please add a test for MethodLengthCheck where checkstyle prints a violation about method length even after deleting lines from a method? This is not what we want, because the method gets smaller, but it is expected for this strategy now.
Something like this:
```diff
void test() { // violation context: max allowed length is 2 lines
  abc();
  abc();
  abc();
-  abc();
}
```

Reply:
done, @baratali I think it is ok to print a violation if it is still over the threshold that set, although user delete some lines from a method.